# Avocado Prices and Sales Volume 2015-2023

In [ ]:
from IPython.display import Image
Image(url='https://www.buygrow.co.za/cdn/shop/products/FuerteAvocadoWB.png?v=1675280423')


# Table of Contents
- [1. Project Overview](#1-project-overview)
  - [1.1 Introduction](#11-introduction)
  - [1.2 Problem Statement](#12-problem-statement)
  - [1.3 Objectives](#13-objectives)
- [2. Importing Packages](#2-importing-packages)
- [3. Loading Data](#3-loading-data)
- [4. Data Cleaning](#4-data-cleaning)
- [5. Exploratory Data Analysis (EDA)](#5-exploratory-data-analysis-eda)
- [6. Conclusion](#6-conclusion)

### 1. Project Overview

##### 1.1 Introduction

This project focuses on data cleaning and exploratory data analysis (EDA) of historical avocado prices. The primary goal is to clean the dataset and explore trends, seasonal patterns, and potential factors affecting avocado prices, setting a foundation for future predictive modeling.

##### 1.2 Problem Statement

##### 1.3 Objectives

- Data Cleaning: Ensure data quality by handling missing values, correcting data types and adding new columns.

- Exploratory Data Analysis (EDA): Identify trends, seasonal variations, and key differences in avocado prices by region and type (conventional vs. organic).

### 2. Importing Packages

To carry out data cleaning, manipulation, and visualization, we’ll use the following Python libraries:

* pandas: Provides data structures and functions needed to efficiently clean and manipulate the dataset.

* numpy: Adds support for numerical operations, including handling arrays and mathematical functions for outlier treatment.

* matplotlib and seaborn: Libraries for data visualization. matplotlib is a core plotting library, while seaborn builds on it to provide more aesthetic and statistical visualizations.

In [2]:
# Libraries for data loading, manipulation and analysis

import numpy as np
import pandas as pd
import csv
import seaborn as sns
import matplotlib.pyplot as plt

# Displays output inline
%matplotlib inline

# Libraries for Handing Errors
import warnings
warnings.filterwarnings('ignore')

### 3. Loading Data

The data used for this project was located in the Avocado_HassAvocadoBoard_20152023v1.0.1.csv file. To better manipulate and analyse the Avocado_HassAvocadoBoard_20152023v1.0.1.csv file, it was loaded into a Pandas Data Frame using the Pandas function, .read_csv(). 

In [3]:
# loading dataset
df = pd.read_csv("Avocado_HassAvocadoBoard_20152023v1.0.1.csv", index_col=False)

Check the DataFrame to see if it loaded correctly.

In [4]:
df.head() 

Date  AveragePrice  TotalVolume    plu4046    plu4225   plu4770  \
0  2015-01-04          1.22     40873.28    2819.50   28287.42     49.90   
1  2015-01-04          1.79      1373.95      57.42     153.88      0.00   
2  2015-01-04          1.00    435021.49  364302.39   23821.16     82.15   
3  2015-01-04          1.76      3846.69    1500.15     938.35      0.00   
4  2015-01-04          1.08    788025.06   53987.31  552906.04  39995.03   

   TotalBags  SmallBags  LargeBags  XLargeBags          type  \
0    9716.46    9186.93     529.53         0.0  conventional   
1    1162.65    1162.65       0.00         0.0       organic   
2   46815.79   16707.15   30108.64         0.0  conventional   
3    1408.19    1071.35     336.84         0.0       organic   
4  141136.68  137146.07    3990.61         0.0  conventional   

                region  
0               Albany  
1               Albany  
2              Atlanta  
3              Atlanta  
4  BaltimoreWashington

In [8]:
# Displays the number of rows and columns
df.shape

(53415, 12)

Results : The dataset consists of 53415 rows (observations) and 12 columns (features).

In [26]:
## Display summary information about the DataFrame.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 53415 entries, 0 to 53414
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Date          53415 non-null  object 
 1   AveragePrice  53415 non-null  float64
 2   TotalVolume   53415 non-null  float64
 3   plu4046       53415 non-null  float64
 4   plu4225       53415 non-null  float64
 5   plu4770       53415 non-null  float64
 6   TotalBags     53415 non-null  float64
 7   SmallBags     41025 non-null  float64
 8   LargeBags     41025 non-null  float64
 9   XLargeBags    41025 non-null  float64
 10  type          53415 non-null  object 
 11  region        53415 non-null  object 
dtypes: float64(9), object(3)
memory usage: 4.9+ MB


In [28]:
df.describe()

AveragePrice   TotalVolume       plu4046       plu4225       plu4770  \
count  53415.000000  5.341500e+04  5.341500e+04  5.341500e+04  5.341500e+04   
mean       1.428910  8.694474e+05  2.982707e+05  2.222170e+05  2.053195e+04   
std        0.393116  3.545274e+06  1.307669e+06  9.554624e+05  1.040977e+05   
min        0.440000  8.456000e+01  0.000000e+00  0.000000e+00  0.000000e+00   
25%        1.119091  1.626465e+04  6.947250e+02  2.120800e+03  0.000000e+00   
50%        1.400000  1.203525e+05  1.458058e+04  1.751663e+04  9.005000e+01   
75%        1.690000  4.542380e+05  1.287924e+05  9.351560e+04  3.599735e+03   
max        3.440830  6.103446e+07  2.544720e+07  2.047057e+07  2.860025e+06   

          TotalBags     SmallBags     LargeBags     XLargeBags  
count  5.341500e+04  4.102500e+04  4.102500e+04   41025.000000  
mean   2.175083e+05  1.039222e+05  2.331316e+04    2731.811796  
std    8.676947e+05  5.692608e+05  1.496622e+05   22589.096454  
min    0.000000e+00  0.000000e+00  0.000000e+00       0.000000  
25%    7.846520e+03  0.000000e+00  0.000000e+00       0.000000  
50%    3.695310e+04  6.945800e+02  0.000000e+00       0.000000  
75%    1.110146e+05  3.795298e+04  2.814920e+03       0.000000  
max    1.629830e+07  1.256716e+07  4.324231e+06  679586.800000

### 4. Data Cleaning

Handling Missing Values

### 5. Exploratory Data Analysis (EDA)

### 6. Conclusion